### ISSUE TITLE
Few questions

### ISSUE THREAD

**Issue by krypt-lynx** [martindevans/Cassowary.net#7]
- how to do constraint priority in this implementation? I see `ClStrength` and `ClSymbolicWeight`, first requires second and ClSymbolicWeight requires 3 numbers to init. What are those numbers?
- I'm trying to wrap my head around and don't see solution yet: how to do nested constraints? 
For example: I have a view inside a view, and I need size of external view is constrained by one set of rules and internal one depends on size of external

Basically, I'm trying to understood it in terms of Apple's AutoLayout

**Comment by martindevans** [comment]
It's a long time since I've used this library, so I'm afraid I can't really remember these kinds of details, sorry :(

The original library which I forked was a port of [this project](https://constraints.cs.washington.edu/cassowary/). In particular [this paper](https://constraints.cs.washington.edu/solvers/cassowary-tochi.pdf) describes the details of the algorithm.

**Comment by krypt-lynx** [comment]
Thanks. Is it that thesis mentioned in original repository?

**Comment by martindevans** [comment]
Yep, it's link #2 in the [readme](https://github.com/jozilla/Cassowary.net/blob/master/README).

**Comment by krypt-lynx** [comment]
Hi! I did quite a few changes in my repo (https://github.com/krypt-lynx/Cassowary.net)
- Operator overrides based constraint builder (looks to be more stable, at least I can make calls in parameters)
- ToString() readability
- Some extra methods in ClSimplexSolver

Not creating merge requests because changes are making my fork incompatible with yours.

Also, looks like the original author cheated in his thesis... In this implementation Strengths are not sorted by order,  like he states in it. So, Medium with weight = 1001 overpowers Strong with weight = 1 :( Also, ClStrengths and Epsilon values appears to be finetuned. (changing them breaks unit tests)
